rfcs/text/0006_management_section_service.md

```diff
@@ -0,0 +1,284 @@
+  };
+}
+
+// ui/management/section
+class ManagementSection {
+  get visibleItems,
+  addListener: (fn: function) => void,
+  register: (id: string, options: Options) => ManagementSection | Error,
+  deregister: (id: string) => void,
+  hasItem: (id: string) => boolean,
+  getSection: (id: string) => ManagementSection,
+  hide: () => void,
+  show: () => void,
+  disable: () => void,
+  enable: () => void,
+}
+ 
+interface Options {
+  order: number | null;
+  display: string | null; // defaults to id
+  url: string | null; // defaults to ''
+  visible: boolean | null; // defaults to true
+  disabled: boolean | null; // defaults to false
+  tooltip: string | null; // defaults to ''
+  icon: string | null; // defaults to ''
+}
+```
+
+# Notes
+
+- The hide/show/disable/enable options were dropped with the assumption that we will be working with uiCapabilities to determine this instead... so people shouldn't need to manage it manually as they can look up a pre-filtered list of sections.
+- This was updated to add flexibility for custom (non-EUI) icons as outlined in #32661. Much like the Core Application Service, you either choose an EUI icon, or provide a URL to an icon.
+
+# Drawbacks
+
+- This removes the ability to infinitely nest sections within each other by making a distinction between a section header and a nav link.
+  - So far we didn't seem to be using this feature anyway, but would like feedback on any use cases for it.
+
+# Unresolved questions
+
+- Do individual links within a section need the ability to specify icons? Currently only the section headers have icons.
+- Do we need a field for a description? Currently there is no place for it in the UI, but it was added based on the mocks from the [reference section](#reference).
+- From a product perspective, what does the future of the Management section look like? There has been recent discussion around making Global vs Space-specific management areas (see links in [reference section](#reference)); if this is a feature we want in an `8.0` timeframe, we will probably need to resolve this question before moving forward. If this is something we plan for `8.x` and beyond, the current proposal remains mostly unaffected and we can add that as 
+
+# Reference
+
+- Issues about Global vs Spaces-based management sections: https://github.com/elastic/kibana/issues/37285 https://github.com/elastic/kibana/issues/37283
+- Mockups related to above issues: https://marvelapp.com/52b8616/screen/57582729
+
+# Alternatives
+
+An alternative design would be making everything React-specific and simply requiring consumers of the service to provide a React component to render when a route is hit, or giving them a react-router instance to work with.
+
+This would require slightly less work for folks using the service as it would eliminate the need for a `mount` function. However, it comes at the cost of forcing folks into a specific rendering framework, which ultimately provides less flexibility.
```

Comment:
> How many registries/extension points are extendable by 3rd party plugins? In the sake of consistency, they should follow the same convention.

Index Management allows other plugins to [extend its UI](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/index_management/public/index_management_extensions.js). Here's an example of these [extension points being consumed](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/index_lifecycle_management/public/extend_index_management/index.js). They're tightly coupled to both React and EUI.

Reply:
> These extension points are tightly coupled to both React and EUI.

**Some of them** are coupled, some use POJO to setup content. Depending on the agreement for management section we can adjust (or not) this API as well.

Reply:
I agree it’s possible to adjust this API but it becomes difficult for us (the ES UI team that owns this code) to justify time spent on refactoring without a clear benefit, since we built these extension points for our own use, not use by third parties.

Reply:
I have no problem with some of these very specific extension points being React. I just see the overall Management app is more generic and I think that should be decoupled from React. I think that the other ES UI extension are very unlikely to be consumed by 3rd parties.